gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageWriteChannel.java

```diff
@@ -83,8 +83,8 @@ public long size() throws IOException {
 
   @Override
   public SeekableByteChannel truncate(long newSize) throws IOException {
-    // TODO(b/18997913): Emulate this functionality by closing and rewriting old file up to newSize.
-    //                   Or maybe just swap out GcsService for the Apiary client.
+    // TODO: Emulate this functionality by closing and rewriting old file up to newSize.
+    //       Or maybe just swap out GcsStorage for the API client.
     throw new UnsupportedOperationException();
   }
 
```

Comment:
Yes, though as our code was changed to throw `ClosedChannelException` when used after it was closed technically you don't need that check in the `write` method.

However, as you need your own `isClosed` for the `size` and `position` method keeping your code as is seems like the right thing to do.

You an still leave it as is for consistency though I do wonder if